replicators/src/mysql_connector/connector.rs

```diff
@@ -128,8 +132,31 @@ impl MySqlBinlogConnector {
         };
 
         connector.register_as_replica().await?;
-        connector.request_binlog().await?;
-
+        let binlog_request = connector.request_binlog().await;
+        match binlog_request {
+            Ok(()) => (),
+            Err(mysql_async::Error::Server(ref err))
+                if mysql_srv::ErrorKind::from(err.code)
+                    == mysql_srv::ErrorKind::ER_MASTER_FATAL_ERROR_READING_BINLOG =>
```

Comment:
> do we know that a ER_MASTER_FATAL_ERROR_READING_BINLOG error will always require a full snapshot for recovery?

Yes, a fatal error should require re-snapshot

> Are there other server errors that would require a full snapshot?

I cannot think of any other. It should be easy to add a new ENUM on the full snapshot list in case we find the need to do so.